Self type with multiple inheritance

Not sure whether this is actually a bug or not, but I'm getting a base class incompatibility error when inheriting from two classes defining a method with the same signature containing the `typing.Self` type.

**To Reproduce**

```python
from typing import Self

class A:
    def a_method(self) -> None:
        pass
    def method(self: Self, other: Self) -> None:
        self.a_method()
        other.a_method()

class B:
    def b_method(self) -> None:
        pass
    def method(self: Self, other: Self) -> None:
        self.b_method()
        other.b_method()

class C(A, B):
    pass
```

Running `mypy` gives this error:

```
test.py:17: error: Definition of "method" in base class "A" is incompatible with definition in base class "B"  [misc]
```

I expected `Self` to mean "whatever this type happens to be" (base class or subclass) but the error would suggest that it's interpreting `Self` as the specific class containing that method definition.

Next I tried the following variation:

```python
from typing import TypeVar

T = TypeVar('T')

class A:
    def a_method(self) -> None:
        pass
    def method(self: T, other: T) -> None:
        self.a_method()

class B:
    def b_method(self) -> None:
        pass
    def method(self: T, other: T) -> None:
        self.b_method()

class C(A, B):
    pass
```

This fixes the incompatibility error but results in new errors:

```
test.py:9: error: "T" has no attribute "a_method"  [attr-defined]
test.py:15: error: "T" has no attribute "b_method"  [attr-defined]
```

I also tried using two separate `TypeVar`s, one with `bound='A'` and the other with `bound='B'`, but this brought back the "incompatible definition" error from before.

Finally (after reverting the types back to `Self`), I updated `C`'s definition to the following:

```python
class C(A, B):
    def method(self: Self, other: Self) -> None:
        A.method(self, other)
```

This ended up type-checking, but I'm wondering if `mypy` ought to be clever enough to figure it out without having to explicitly override `method`, or if there's some other alternative?

And if not, perhaps the error message about the incompatible definition could elaborate a bit more in the case where the type signatures "appear" to match due to the use of `Self`?

**Environment**

- Python 3.11.11
- `mypy` 1.14.1 (no flags or config file)

--- a/mypy/checker.py
+++ b/mypy/checker.py
@@ -2232,8 +2232,8 @@ class TypeChecker(NodeVisitor[None], CheckerPluginInterface):
                 is_class_method = sym.node.is_class
 
             mapped_typ = cast(FunctionLike, map_type_from_supertype(typ, sub_info, super_info))
-            active_self_type = self.scope.active_self_type()
-            if isinstance(mapped_typ, Overloaded) and active_self_type:
+            active_self_type = fill_typevars(sub_info)
+            if isinstance(mapped_typ, Overloaded):
                 # If we have an overload, filter to overloads that match the self type.
                 # This avoids false positives for concrete subclasses of generic classes,
                 # see testSelfTypeOverrideCompatibility for an example.

--- a/test-data/unit/check-selftype.test
+++ b/test-data/unit/check-selftype.test
@@ -2214,3 +2214,22 @@ class Test2:
 
 reveal_type(Test2().method) # N: Revealed type is "def (foo: builtins.int, *, bar: builtins.str) -> builtins.bytes"
 [builtins fixtures/tuple.pyi]
+
+[case testSelfInMultipleInheritance]
+from typing_extensions import Self
+
+class A:
+    foo: int
+    def method(self: Self, other: Self) -> None:
+        self.foo
+        other.foo
+
+class B:
+    bar: str
+    def method(self: Self, other: Self) -> None:
+        self.bar
+        other.bar
+
+class C(A, B):  # OK: both methods take Self
+    pass
+[builtins fixtures/tuple.pyi]

